integration/spark/vendor/snowflake/src/main/java/io/openlineage/spark/agent/vendor/snowflake/lifecycle/plan/QueryPlanDatasetBuilder.java

```diff
@@ -0,0 +1,81 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.vendor.snowflake.lifecycle.plan;
+
+import static io.openlineage.spark.agent.vendor.snowflake.SnowflakeVendor.hasSnowflakeClasses;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.agent.util.PlanUtils;
+import io.openlineage.spark.agent.util.ScalaConversionUtils;
+import io.openlineage.spark.agent.vendor.snowflake.lifecycle.SnowflakeDataset;
+import io.openlineage.spark.api.AbstractQueryPlanDatasetBuilder;
+import io.openlineage.spark.api.OpenLineageContext;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import net.snowflake.spark.snowflake.DefaultSource;
+import org.apache.spark.scheduler.SparkListenerEvent;
+import org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand;
+import org.apache.spark.sql.sources.CreatableRelationProvider;
+import org.apache.spark.sql.types.StructType;
+
+/** Look to {@link io.openlineage.spark.agent.lifecycle.plan.SaveIntoDataSourceCommandVisitor} */
+public class QueryPlanDatasetBuilder
```

Comment:
Maybe rename it `SnowflakeSaveIntoDataSourceCommandDatasetBuilder`?

Reply:
Done